### ISSUE TITLE
matplotlib cutsom style sheet: format dates with new line character

### ISSUE THREAD

**Issue by blaylockbk** [matplotlib/matplotlib#15813]
### Bug report

**Bug summary**

In my custom matplotlib style sheet, I want to format the dates in such a way that includes a new line character, i.e., `%b %d\n%H:%M`. But matplotlib will not recognize the new line character and instead prints `\n` as text on my figure.

More details here: https://stackoverflow.com/questions/59144140/matplotlib-cutsom-style-sheet-format-dates-with-new-line-character?noredirect=1#comment104516325_59144140

**Code for reproduction**

In my `my_style.mplstyle` file, I have this formatting instruction for the dates (note the new line character):

```python
    date.autoformatter.hour    : %b %d\n%H:%M
```

When I load the style sheet and make a plot with dates on the x axis, it displays the \n as text rather than recognizing it as a new line character.

```python
    import matplotlib.pyplot as plt
    plt.style.use('my_style')
    import pandas as pd
    from datetime import datetime
    DATES = pd.date_range(datetime(2019, 1, 1), datetime(2019, 1, 2), 6)
    plt.plot(DATES, range(len(DATES)))
```

**Actual outcome**

![image](https://user-images.githubusercontent.com/6249613/69989378-5775bc00-14f8-11ea-8140-2da354c390a4.png)

```
# If applicable, paste the console output here
#
#
```

**Expected outcome**

    
I can get the desired outcome if I use the manual method to change the date formats directly in my code, but I would rather avoid repeating these lines of code in all my plotting scripts. That is why I want to use the style sheet to do this.

```python
    import matplotlib.dates as mdates
    myFmt = mdates.DateFormatter('%b %d\n%H:%M')
    plt.plot(DATES, range(len(DATES)))
    plt.gca().xaxis.set_major_formatter(myFmt)
```

![image](https://user-images.githubusercontent.com/6249613/69989465-7f651f80-14f8-11ea-8c09-c4af591745ec.png)

**Matplotlib version**

  * Operating system: Windows
  * Matplotlib version: 3.1.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipykernel.pylab.backend_inline
  * Python version: 3.7.4
  * Jupyter version (if applicable): 
  * Other libraries: 

Installed from conda

**Comment by timhoffm** [comment]
This is currently not possible because the parser interprets values as literal strings and thus the backslash is an actual backslash.
~~~
In [2]: plt.rcParams['date.autoformatter.hour']                                        
Out[2]: '%b %d\\n%H:%M'
~~~

Maybe we should do `val.decode('unicode_escape')` but that would be an API change.

**Comment by jklymak** [comment]
That proposal seems reasonable to me

**Comment by anntzer** [comment]
> We could however optionally allow string quotes `date.autoformatter.hour : "%b %d\n%H:%M"` and only text in quotes will be unescaped. That would keep backward-compatibility.

I don't think this keeps backcompat?  I'd say the breakage is unlikely to matter in practice, but both adding support for quotes and adding support for escapes break backcompat to similar extents...

Perhaps a better solution would be to just use a different syntax to mark "new" entries, e.g.
```
date.formatter.hour = "..."
```
i.e. the equals sign (rather than colon) says that what comes after is a python literal (which we should just load using `ast.literal_eval`).

**Comment by timhoffm** [comment]
Theoretically, it's backward compatible, but only if somebody had used string-quoting before and meaning the quotes literally. The likelyhood lf that is small, and I'd be willing to risk that. Since we will only parse escapes in quotes the breakage probability of that is ​ɛ​².

I'm not convinced of making a semantic difference with `=` vs. `:`. This is a good recipe to confuse users.

I'd also be open towards making the config YAML, which AFAICS the current config is a subset of (PyYAML has become active again for what it's worth).
I'm alternatively not completely opposed to a python config, but would need a little more convincing that's a benefit worth introducing a large incompatibility (or us maintaining two formats, respectively.

**Comment by anntzer** [comment]
If we choose to move to yaml (ugh) or python, the easiest is likely to just make the config files `matplotlibrc.yml` or `matplotlibrc.py` respectively, which will allow supporting both formats during the transition period without any guessing/incompatibility (and even after we drop (perhaps) support for old-style matplotlibrc, people can just keep both formats around if they want backcompat with old matplotlibs).

**Comment by QuLogic** [comment]
Quoted strings were added in #22589. @timhoffm intended to add support for backslash escapes, but it doesn't look like that has been implemented yet.
